scalanet/src/io/iohk/scalanet/peergroup/SimplePeerGroup.scala

```
@@ -45,12 +46,12 @@ class SimplePeerGroup[A, AA, M](
     underlyingChannels.map(new ChannelImpl(to, _))
   }
 
-  override def server(): Observable[Channel[A, M]] = {
-    underLyingPeerGroup.server().map { underlyingChannel: Channel[AA, Either[ControlMessage[A, AA], M]] =>
-      val reverseLookup: mutable.Map[AA, A] = routingTable.map(_.swap)
-      val a = reverseLookup(underlyingChannel.to)
-      debug(s"Received new server channel from $a")
-      new ChannelImpl(a, List(underlyingChannel))
+  override def server(): Observable[ServerEvent[A, M]] = {
+    underLyingPeerGroup.server().collectChannelCreated.map {
```

Are you using just channel creations, silently ignoring other server events? Is it possible to expose handshake failures somehow?

---

True. I've also mapped the handshake failures.